server/main.go

```diff
@@ -38,6 +38,10 @@ func getWebHandler(activeFileManager *ActiveFileManager, fileStore FileStore) ht
 		method := req.Method
 		path := urlPathToArray(req.URL.Path)
 
+		res.Header().Set("Access-Control-Allow-Origin", "*")
```

Comment:
So I realize you're trying to XHR this, but why do we need cross-domain XHRs? Could we have the InstantShare server serve the instant upload page?